src/main/scala/diplomaticobjectmodel/logicaltree/LogicalTrees.scala

```diff
@@ -97,3 +99,71 @@ class SubSystemLogicalTreeNode(var getOMInterruptDevice: (ResourceBindingsMap) =
     )
   }
 }
+
+case class OMSRAMData[T <: Data](
+  data: T,
+  resourceBindings: ResourceBindings,
+  architecture: RAMArchitecture,
+  description: String,
+  depth: Int,
+  ecc: Option[OMECC] = None,
+  hasAtomics: Option[Boolean] = None
+)
+
+class SRAMLogicalTreeNode[T <: Data](description: String, architecture: RAMArchitecture, depth: Int, lanes: Int, bits: Int, device: () => Device,
```

Comment:
Why does this class need both `rbm: () => ResourceBindingsMap` AND `def getOMComponents(resourceBindingsMap: ResourceBindingsMap,...)` Are these the same RBM, or different RBMs? Something seems wrong about the fact that there are two and you only use one...